ts/src/config.ts

```diff
@@ -61,25 +61,39 @@ export interface Config extends AuthenticationConfig {
   disableHeap?: boolean;
 }
 
-/**
- * Subtype of Config that declares some properties to be non-optional. This
- * helps avoid null checks in the profiler code.
- */
-export interface ProfilerConfig extends Config {
+// Interface for an initialized config.
+export interface ProfilerConfig extends AuthenticationConfig {
+  projectId?: string;
   logLevel: number;
-  serviceContext: {service?: string; version?: string;};
+  serviceContext: {service: string; version?: string;};
   instance: string;
   zone: string;
   disableTime: boolean;
   disableHeap: boolean;
+
+  // Internal configuration parameters.
+  // Modifying values for these fields is not recommended.
+  timeIntervalMicros: number;
+  heapIntervalBytes: number;
+  heapMaxStackDepth: number;
+  backoffMillis: number;
 }
 
 // Default values for configuration for a profiler.
-export const defaultConfig: ProfilerConfig = {
+export const defaultConfig = {
   logLevel: 1,
   serviceContext: {},
   disableHeap: false,
   disableTime: false,
   instance: '',
   zone: ''
 };
+
+// Configuration parameters set internally.
+// Modifying these values is not recommended.
+export const internalConfig = {
```

Comment:
I would suggest adding it to defaultConfig. That allows up to tweak/experiment without making source changes to the module. Everything in JS is public anyway, but this way is more convenient:

```ts
export const defaultConfig = {
  logLevel: 1,
  serviceContext: {},
  disableHeap: false,
  disableTime: false,
  instance: '',
  zone: '',
  internal: {
    timeIntervalMicros: 1000,
    heapIntervalBytes: 512 * 1024,
    heapMaxStackDepth: 64,
    backoffMillis: 5 * 60 * 1000,  // 5 minutes
  } 
};
```

This will also force you to declare types for these, and document them. That is a Good Thing even for internal configuration, IMO.